source/Account.cpp

```diff
@@ -191,20 +197,40 @@ string Account::Step(int64_t assets, int64_t salaries)
 			++it;
 	}
 	
+	// Maintenance costs are dealt with last since missing maintenance costs do
+	// not impact your credit score.
+	int64_t maintenancePaid = maintenanceDue;
+	if(maintenanceDue)
+	{
+		if(maintenanceDue > credits)
+		{
+			maintenancePaid = max<int64_t>(credits, 0);
+			maintenanceDue -= maintenancePaid;
+			credits -= maintenancePaid;
+			if(!missedPayment)
+				out << "You could not pay all your maintenance costs.";
```

Comment:
Add `missedPayment = true;` here, similar to salaries and morgage above.

Reply:
I've omitted that because overdue maintenance costs don't impact credit score, and if missedPayment being true is what causes your credit score to decrease.

Reply:
I think it's okay if overdue maintenance costs impact the credit score. You pay something too late (in this case maintenance) -> you get a lower credit score.

Reply:
With the idea that [you're borrowing to pay for maintenance](https://github.com/endless-sky/endless-sky/issues/4284#issuecomment-541769571) if you don't have the credits on hand, or that operating costs are [to account for more expensive crew](https://github.com/endless-sky/endless-sky/issues/4284), I guess it isn't out of the question that overdue maintenance would impact your credit score.